### ISSUE TITLE
Support getting authoritative nameservers for a zone

### ISSUE THREAD

**Issue by F21** [G-Core/gcore-dns-sdk-go#1]
There should be a method to get the authoritative name servers for a zone.
API endpoint: https://apidocs.gcore.com/dns#tag/zones/operation/AnalyzeZone
This would allow the implementation of https://github.com/G-Core/terraform-provider-gcorelabs/issues/126

**Comment by clickfreak** [comment]
Do you mean the addresses to which you need to delegate the domain? ("Please delegate your domain on following Gcore nameservers: <list of NS>")

Or get current nameservers? Example:
```
$ curl -s -H "Authorization: ..." "https://api.gcore.com/dns/v2/analyze/boostport.com/delegation-status" | jq .                      
{
  "gcore_authorized_count": 0,
  "non_gcore_authorized_count": 12,
  "authoritative_name_servers": [
    {
      "name": "macy.ns.cloudflare.com.",
      "ipv4Addresses": [
        "108.162.194.78",
        "162.159.38.78",
        "172.64.34.78"
      ],
      "ipv6Addresses": [
        "2606:4700:50::a29f:264e",
        "2803:f800:50::6ca2:c24e",
        "2a06:98c1:50::ac40:224e"
      ]
    },
    {
      "name": "karl.ns.cloudflare.com.",
      "ipv4Addresses": [
        "108.162.193.190",
        "172.64.33.190",
        "173.245.59.190"
      ],
      "ipv6Addresses": [
        "2606:4700:58::adf5:3bbe",
        "2803:f800:50::6ca2:c1be",
        "2a06:98c1:50::ac40:21be"
      ]
    }
  ],
  "zone_exists": true,
  "is_whitelabel_delegation": false
}
```

**Comment by F21** [comment]
I would like to retrieve the G-core labs servers to delegate my domain to (`ns1.gcorelabs.net` and `ns2.gcdn.services`):
![Untitled-2](https://user-images.githubusercontent.com/2263040/210125989-81b2984d-0fc9-459f-97cc-6a08d159468e.png)

**Comment by rrickardt** [comment]
Do i understand that correctly that you are not willing to hardcode ns1.gcorelabs.net and ns2.gcdn.services to your code and need some API service to return you these 2 nameservers?

**Comment by F21** [comment]
@rrickardt Yes, that's what I am hoping to achieve. If this library supports that, I can add create a PR in https://github.com/G-Core/terraform-provider-gcorelabs to use it to return the nameservers, which can be used to assign nameservers to domains with terrraform without hardcoding the 2 G-Core Labs name servers.

**Comment by rrickardt** [comment]
@F21 ok, as i see in API https://apidocs.gcore.com/dns#tag/zones/operation/CreateZone primary_server parameter is supported in API, when you GET Zone, there is primary_server returned as well. So far, i see it as supported in API, but not in client. So you are looking for sdk support for primary_server parameter?

**Comment by F21** [comment]
The problem is that `primary_server` only returns the primary server `ns1.gcorelabs.net` and does not include the secondary server. It seems that the `analyze` endpoint mentioned in the original issue is no longer shown at https://apidocs.gcore.com/dns, but appears to work. Not too sure if it is safe to rely on that undocumented endpoint.

**Comment by rrickardt** [comment]
To get servers from API you can use https://apidocs.gcore.com/dns#tag/zones/operation/Zone
endpoint. It should work from SDK as well.

That way you get 

"records":[{"name":"blahbee.no","type":"NS","ttl":3600,"short_answers":["ns1.gcorelabs.net","ns2.gcdn.services"]}]

And you check for NS records. Does it work for you please?

I do not think there is still currently  possibility to specify 2 nameservers on zone creation. https://apidocs.gcore.com/dns#tag/zones/operation/CreateZone

primary_server is used to specify SOA record. Documentation is not clear on this. 

I am trying it with free plan and i can not specify my own nameservers. Are you on paid one?

**Comment by F21** [comment]
I am also on the free plan, so I cannot test custom name servers. Not sure if looking up the NS records are the best way to get the nameservers. In AWS' Route53, getting a zone also returns the [delegation set](https://docs.aws.amazon.com/Route53/latest/APIReference/API_GetHostedZone.html#API_GetHostedZone_Examples) so it is very clear and unambigious as to what the authoritative DNS servers are, rather than needing to parse the NS records.

For example, after creating a zone, it is possible to add an NS record to it. Getting the zone results in:

```
{"name":"testdomaincom","meta":null,"enabled":true,"nx_ttl":300,"retry":3600,"refresh":0,"expiry":1209600,"contact":"support@gcorelabs.com","serial":1674213632,"primary_server":"ns1.gcorelabs.net","records":[{"name":"boostport.com","type":"NS","ttl":3600,"short_answers":["ns1.gcorelabs.net","ns2.gcdn.services","sometest.com"]}],"status":"non-delegated"}
```

This makes it really brittle if we rely on the `short_answers` key to parse the delegated g-core name servers.

**Comment by rrickardt** [comment]
On AWS example, these are responses on GET requests. 

So for Gcore DNS API, you suggest to have a response body based on POST request with some zone data? Or some another endpoint that returns lets say SOA and NS records, delegation status etc. based on GET request and some parameter?

**Comment by F21** [comment]
I am suggesting returning the delegated nameservers via GET. Whether it's with the zone data, or some other request doesn't matter, but being able to retrieve the delegated nameservers would bea  huge benefit.

**Comment by rrickardt** [comment]
@F21 please check https://github.com/G-Core/gcore-dns-sdk-go/pull/3
